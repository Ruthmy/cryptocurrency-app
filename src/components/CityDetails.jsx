import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';

import weatherIcons from '../assets/bd/weatherIcons';
import '../styles/CityDetails.css';

const CityDetails = () => {
  const { city } = useParams();

  // Get the current conditions from the global state
  const CityDetails = useSelector((state) => state.weather.weather).find(
    (country) => country.TimeZone.Name.includes(city),
  );

  return (
    <div>
      <div className="home d-flex-row">
        <img
          className="header"
          src={weatherIcons[CityDetails.WeatherIcon]}
          alt={CityDetails.WeatherText}
        />
        <div className="home__info">
          <h1 className="title">{city}</h1>
          <p className="description" />
        </div>
      </div>
      <div className="division">
        <h2>
          CURRENT CONDITIONS
        </h2>
      </div>
      {CityDetails && (
        <div className="weather d-flex-column">

          <div className="weather__details d-flex-column">
            <p className="temperature">
              {CityDetails.Temperature.Metric.Value}
              &nbsp;
              {CityDetails.Temperature.Metric.Unit}
              ° |&nbsp;
              {CityDetails.Temperature.Imperial.Value}
              &nbsp;
              {CityDetails.Temperature.Imperial.Unit}
              °
            </p>
            <div className="summary">
              <p className="summaryText">
                {CityDetails.WeatherText}
              </p>
            </div>
          </div>
        </div>
      )}
    </div>
  );
};

export default CityDetails;
