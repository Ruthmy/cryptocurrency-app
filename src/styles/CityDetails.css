.weather {
  align-items: center;
  justify-content: center;
  padding: 1rem;
  width: 100%;
}

.weather__details {
  width: 100%;
  margin-bottom: 2rem;
  font-family: var(--font-one);
}

.temperature {
  color: white;
  font-weight: 300;
  font-size: 5.5em;
}

.summary {
  width: fit-content;
  color: #d2e9fa;
  padding-bottom: 0.5rem;
  border-bottom: 2px solid #9cd0ff;
}

.summaryText {
  font-size: 3em;
  font-weight: 300;
}

.weather__aditional {
  align-items: flex-start;
  width: 100%;
  gap: 1rem;
}
