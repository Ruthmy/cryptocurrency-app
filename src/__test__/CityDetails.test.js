import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom';

import CityDetails from '../components/CityDetails';

const mockStore = configureStore([]);
let store;

// Helper function to wrap the component in MemoryRouter for testing NavLink
const renderWithRouter = (component) => render(
  <Provider store={store}>
    <MemoryRouter initialEntries={['/Asia/Dhaka']}>
      <Routes>
        <Route path="/:continent/:city" element={component} />
      </Routes>
    </MemoryRouter>
  </Provider>,
);

describe('Continent component', () => {
  beforeEach(() => {
    store = mockStore({
      weather: {
        weather: [{
          Key: '28143',
          LocalizedName: 'Dhaka',
          EnglishName: 'Dhaka',
          Country: { ID: 'BD', LocalizedName: 'Bangladesh', EnglishName: 'Bangladesh' },
          TimeZone: {
            Code: 'BDT', Name: 'Asia/Dhaka', GmtOffset: 6, IsDaylightSaving: false, NextOffsetChange: null,
          },
          GeoPosition: { Latitude: 23.7098, Longitude: 90.40711, Elevation: { Metric: { Value: 5, Unit: 'm', UnitType: 5 }, Imperial: { Value: 16, Unit: 'ft', UnitType: 0 } } },
          LocalObservationDateTime: '2023-08-05T08:37:00+06:00',
          EpochTime: 1691203020,
          WeatherText: 'Cloudy',
          WeatherIcon: 7,
          HasPrecipitation: false,
          PrecipitationType: null,
          IsDayTime: true,
          Temperature: { Metric: { Value: 28.5, Unit: 'C', UnitType: 17 }, Imperial: { Value: 83, Unit: 'F', UnitType: 18 } },
          MobileLink: 'http://www.accuweather.com/en/bd/dhaka/28143/current-weather/28143?lang=en-us',
          Link: 'http://www.accuweather.com/en/bd/dhaka/28143/current-weather/28143?lang=en-us',
        }],
        status: 'succeeded',
        error: null,
      },
    });
  });

  test('Should have the city name in the header/home section', () => {
    const { getByText } = renderWithRouter(<CityDetails />);
    const titleElement = getByText('Dhaka');
    expect(titleElement).toBeInTheDocument();
  });

  test('Should have render the division section', () => {
    renderWithRouter(<CityDetails />);
    const divisionElement = screen.getByText('CURRENT CONDITIONS');
    expect(divisionElement).toBeInTheDocument();
  });

  test('Should have the current conditions image', () => {
    renderWithRouter(<CityDetails />);
    const continentImageIsPresent = screen.getByAltText('Cloudy');
    expect(continentImageIsPresent).toBeInTheDocument();
  });

  test('Should have the city Name and its temperature', () => {
    renderWithRouter(<CityDetails />);
    const temperatureIsPresent = screen.getByText('28.5 C° | 83 F°');
    expect(temperatureIsPresent).toBeInTheDocument();
  });

  test('Should render correctly', () => {
    const { container } = renderWithRouter(<CityDetails />);
    expect(container).toMatchSnapshot();
  });
});
