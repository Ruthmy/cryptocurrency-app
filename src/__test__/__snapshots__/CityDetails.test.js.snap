// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Continent component Should render correctly 1`] = `
<div>
  <div>
    <div
      class="home d-flex-row"
    >
      <img
        alt="Cloudy"
        class="header"
        src="https://www.accuweather.com/images/weathericons/7.svg"
      />
      <div
        class="home__info"
      >
        <h1
          class="title"
        >
          Dhaka
        </h1>
        <p
          class="description"
        />
      </div>
    </div>
    <div
      class="division"
    >
      <h2>
        CURRENT CONDITIONS
      </h2>
    </div>
    <div
      class="weather d-flex-column"
    >
      <div
        class="weather__details d-flex-column"
      >
        <p
          class="temperature"
        >
          28.5
           
          C
          ° | 
          83
           
          F
          °
        </p>
        <div
          class="summary"
        >
          <p
            class="summaryText"
          >
            Cloudy
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
